### ISSUE TITLE
Specify CultureInfo in Controller make query not working

### ISSUE THREAD

**Issue by utarn** [dotnet/efcore#18831]
I want to force the user UI to be in Thai language. So, I set cultureinfo using this code. 

``` C#
var cultureInfo = CultureInfo.GetCultureInfo("th-TH");
Thread.CurrentThread.CurrentCulture = cultureInfo;
Thread.CurrentThread.CurrentUICulture = cultureInfo;
```

I also tried app.UseRequestLocalization and added only "th-TH" supported.
It's working fine on windows. But not in linux. I have tried running on asp.net core running docker image from mcr.microsoft.com. Still, it didn't work. I also tried on CentOS within and outside the container. It produces the same issue.

After adding this code, on Windows, the date time is rendered in Thai format properly. When running kestrel on linux and the query with parameter (Where clause) is specified in LINQ, it throws an exception.

 PostgresException: 42703: column "__normalizedusername_0" does not exist

### Steps to reproduce
Add this code. 

``` C#
var cultureInfo = CultureInfo.GetCultureInfo("th-TH");
Thread.CurrentThread.CurrentCulture = cultureInfo;
Thread.CurrentThread.CurrentUICulture = cultureInfo;
```

and try running 

``` C#
await _userManager.FindByNameAsync(username_to_test);
```

Output on /Identity/Account/Login

```
Npgsql.NpgsqlConnector+<>c__DisplayClass160_0+<<DoReadMessage>g__ReadMessageLong|0>d.MoveNext()
Npgsql.NpgsqlConnector+<>c__DisplayClass160_0+<<DoReadMessage>g__ReadMessageLong|0>d.MoveNext()
Npgsql.NpgsqlDataReader.NextResult(bool async, bool isConsuming)
Npgsql.NpgsqlCommand.ExecuteReaderAsync(CommandBehavior behavior, bool async, CancellationToken cancellationToken)
Npgsql.NpgsqlCommand.ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
Microsoft.EntityFrameworkCore.Query.RelationalShapedQueryCompilingExpressionVisitor+AsyncQueryingEnumerable<T>+AsyncEnumerator.MoveNextAsync()
System.Runtime.CompilerServices.ValueTaskAwaiter<TResult>.GetResult()
Microsoft.EntityFrameworkCore.Query.ShapedQueryCompilingExpressionVisitor.SingleOrDefaultAsync<TSource>(IAsyncEnumerable<TSource> asyncEnumerable, CancellationToken cancellationToken)
Microsoft.EntityFrameworkCore.Query.ShapedQueryCompilingExpressionVisitor.SingleOrDefaultAsync<TSource>(IAsyncEnumerable<TSource> asyncEnumerable, CancellationToken cancellationToken)
Microsoft.AspNetCore.Identity.UserManager<TUser>.FindByNameAsync(string userName)
Microsoft.AspNetCore.Identity.SignInManager<TUser>.PasswordSignInAsync(string userName, string password, bool isPersistent, bool lockoutOnFailure)
eSignatureWebApp.Areas.Identity.Pages.Account.LoginModel.OnPostAsync(string returnUrl) in Login.cshtml.cs
```

### Further technical details

EF Core version: 3.0.0
Database provider: Npgsql 4.1.1 and Npgsql.EntityframeworkCore.Postgresql 3.0.1
Target framework: .net core 3.0
Operating system: CentOS 7 within container, CentOS 7, container mcr.microsoft.com/dotnet/core/aspnet:3.0
IDE: Visual Studio 2019

**Comment by ajcvickers** [comment]
@roji Any thoughts here? Could this be a Linux config issue?

**Comment by roji** [comment]
@utarn could you please submit a minimal project triggering the issue?

**Comment by utarn** [comment]
@roji I'm trying to create a new simple project but still not able to reproduce the issue. Please wait.

**Comment by utarn** [comment]
I'm have not managed to reproduce the error. I put it here. https://github.com/utarn/testculture. Also the docker-compose file I use to test.

**Comment by roji** [comment]
Confirmed this is a bug in EF Core, submitted fix.

**Comment by mbp** [comment]
We have this issue in production. We would really like this fix in 3.1. Why is it needed to wait until 5.0? Clearly a bug. It did work in 2.2.

**Comment by ErikEJ** [comment]
Can you implement a workaround in Postgres provider, @roji ?

**Comment by roji** [comment]
@ajcvickers what do you think?

**Comment by roji** [comment]
At the very worst, it's possible to work around this by extending the SqlGenerationHelper of the provider you're using and override GenerateParameterName.

**Comment by mbp** [comment]
I have implemented suggested work-around and it appears to work

Will show solution here if anyone else needs it. But I do believe this change would be nice to get into mainline, seems really trivial fix.

```
public class OrdinalCompareNpgsqlGenerationHelper : NpgsqlSqlGenerationHelper
{
    public OrdinalCompareNpgsqlGenerationHelper(RelationalSqlGenerationHelperDependencies dependencies) 
         : base(dependencies)
    {
    }

    public override string GenerateParameterName(string name)
        => name.StartsWith("@", StringComparison.Ordinal)
            ? name
            : "@" + name;
}
```
```
var dbContextOption = new DbContextOptionsBuilder<T>()
    .UseNpgsql(dbConnectionOptions.ConnectionString)
    .ReplaceService<ISqlGenerationHelper, OrdinalCompareNpgsqlGenerationHelper>()
```

**Comment by roji** [comment]
We already fixed this in the 5.0 branch, so the first 5.0 preview will already contain this fix. At the moment we don't plan to patch this in 3.x, since the workaround is very easy and not many people are affected.

**Comment by smitpatel** [comment]
PR not merged yet.

**Comment by ultimaweapon** [comment]
When 3.1.11 that contains this fix will be released? I have some reasons that cannot move to 5.0. If it is not going to release soon I will use workaround instead.

**Comment by wtgodbe** [comment]
> When 3.1.11 that contains this fix will be released?

Tomorrow (January 12th)

**Comment by roji** [comment]
@Tim1104 are you seeing this issue when using EF Core 5.0? If so, can you please open a new issue with a runnable code sample that shows the bug you're seeing?

**Comment by Tim1104** [comment]
did this issue fixed in 3.1.11?

**Comment by moghthalkushal** [comment]
@roji   Is the fix applicable to SQL Server too?
